message/validation/validation.go

```diff
@@ -96,6 +97,12 @@ func (mv *messageValidator) Validate(_ context.Context, peerID peer.ID, pmsg *pu
 
 	decodedMessage, err := mv.handlePubsubMessage(pmsg, time.Now())
 	if err != nil {
+		if mv.netCfg.Beacon.EstimatedCurrentSlot() == mv.netCfg.Beacon.FirstSlotAtEpoch(mv.netCfg.AlanForkEpoch) {
+			fullErr, ok := err.(Error)
```

Comment:
This should work, but if we do a refactor that wraps ErrMalformedPubSubMessage into another error (we'll unlikely do it though), the check will fail and this will be complex to find out. I'd consider using `errors.As` instead